CRM/Upgrade/Form.php

```diff
@@ -624,6 +631,33 @@ public static function doFileCleanup(CRM_Queue_TaskContext $ctx, $postUpgradeMes
     return TRUE;
   }
 
+  /**
+   * Disable any extensions not compatible with this new version.
+   *
+   * @param \CRM_Queue_TaskContext $ctx
+   * @param string $postUpgradeMessageFile
+   * @return bool
+   */
+  public static function disableOldExtensions(CRM_Queue_TaskContext $ctx, $postUpgradeMessageFile) {
+    $compatInfo = CRM_Extension_System::getCompatibilityInfo();
+    $disabled = [];
+    $manager = CRM_Extension_System::singleton()->getManager();
+    foreach ($compatInfo as $key => $ext) {
+      if (!empty($ext['obsolete']) && $manager->getStatus($key) == $manager::STATUS_INSTALLED) {
+        $disabled[$key] = sprintf("<li>%s</li>", ts('The extension %1 is now obsolete and has been disabled.', [1 => $key]));
+      }
+    }
+    if ($disabled) {
+      $manager->disable(array_keys($disabled));
+      file_put_contents($postUpgradeMessageFile,
+        '<br/><br/><ul>' . implode("\n", $disabled) . '</ul>',
```

Comment:
JFYI, this looks a bit weird when juxtaposed with other messages:

![image](https://user-images.githubusercontent.com/1336047/50531257-cf06c400-0abb-11e9-837a-ea3145613fb2.png)

Reply:
It's the same format used by the `doFileCleanup` fn above.

Reply:
Ah, that makes sense that there are conflicting examples. The issue is probably bigger than this one upgrade step, so we don't need to resolve it right here.